src/App.tsx

```diff
@@ -1,29 +1,47 @@
 import * as React from "react";
-import { BrowserRouter as Router, Route, Link } from "react-router-dom";
-import styled from "styled-components";
-import {GlobalStyle} from "./global";
-import "./App.css";
-import NavBar from "./components/NavBar/NavBar"
-import Home from "./pages/Home/Home"
-import Restaurants from "./pages/Restaurants/restaurantsList"
-import dishPreview from "./pages/Restaurants/dishPreview"
-import Footer from "./components/Footer/footer"
+import { useState, useEffect } from "react";
 
+import { BrowserRouter as Router, Route } from "react-router-dom";
+import firebase from "./firebase";
+// import styled from "styled-components";
+import { GlobalStyle } from "./global";
+import "./App.css";
+import NavBar from "./components/NavBar/NavBar";
+import Home from "./pages/Home/Home";
+import Restaurants from "./pages/Restaurants/restaurantsList";
+import Dishes from "./pages/Dishes/DishList";
+// import dishPreview from "./pages/Restaurants/dishPreview";
+import Footer from "./components/Footer/footer";
 
 function App() {
+  const [restaurants, setRestaurants] = useState([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const db = firebase.firestore();
+      let restaurantsRef = db.collection('restaurants');
+      let allRestaurants = await restaurantsRef.get()
+      let dataArray:any = []
+      for (const doc of allRestaurants.docs){
+        dataArray.push({...doc.data(), id: doc.id})
+      }
+      setRestaurants(dataArray)
```

Comment:
What would happen if you have 10,000 restaurants in your database? would you always want to fetch them all?
And as a follow up, you would have to do this for dishes as well and so on...

Reply:
That's a very good point. 
If the app had a big database I would have probably limit the request and use pagination.